numpy/core/src/multiarray/methods.c

```diff
@@ -2153,7 +2157,10 @@ array_setstate(PyArrayObject *self, PyObject *args)
             Py_RETURN_NONE;
         }
         /* Store the functions in case the global handler is modified */
-        fa->mem_handler = current_handler;
+        fa->mem_handler = (PyDataMem_Handler *) PyDataMem_GetHandler(NULL);
```

Comment:
We need the cast because of the `const`? Maybe it would be better to declare `mem_handler` with a `const` qualifier instead?

Reply:
True. I think we should better `const` the struct field.

Reply:
On a second look, since we `tp_alloc` the `PyArrayObject_fields` struct, making the field `const` would also require a cast to set it.

```
*(PyDataMem_Handler *)&fa->mem_handler = PyDataMem_GetHandler(NULL);
```
I believe it's best to keep it as is. What do you think?